### PR TITLE
Fix (seeding) Randomness utility

### PR DESCRIPTION
**Pull request by svij-sc**
Updating random --> determinisitc utility:
1. Remove TF as we are not using it anymore
2. Simplify defaults and update what is being called according to recent dependency updates.

### CODE REVIEW

File: gigl/common/utils/compute/random.py [hunk 1]
```diff
@@ -1,49 +1,69 @@
...
+      (``torch.manual_seed`` calls ``torch.cuda.manual_seed_all`` internally.
+      Also covers PyTorch Geometric.
 
-    if should_consider_numpy:
-        import numpy as np
+    When ``should_enable_expensive_deterministic_compute=True`` (opt-in; default False because it costs
+    throughput and should not be enabled for training or for production inference - can be used for debugging purposes.
 
-        np.random.seed(seed)
+    - Important: Graph Sampling currently do not follow determism outlined here.
```
Comment (mkolodner-sc) [inline gigl/common/utils/compute/random.py]: nit: determism

File: gigl/common/utils/compute/random.py [hunk 2]
```diff
@@ -1,49 +1,69 @@
...
+        deterministic: If True, also enforces bitwise-deterministic torch
+            ops (cudnn flags, ``use_deterministic_algorithms``,
+            ``CUBLAS_WORKSPACE_CONFIG``). Default False — most training
+            pipelines want seeded RNGs without paying the throughput cost.
 
-        torch.manual_seed(seed)
+    """
+    random.seed(seed)
+    np.random.seed(seed)
+    torch.manual_seed(seed)
```
Comment (mkolodner-sc) [inline gigl/common/utils/compute/random.py:57]: Should we also do `torch.cuda.manual_seed_all(seed)` as is done [here](https://github.com/alibaba/graphlearn-for-pytorch/blob/88ff111ac0d9e45c6c9d2d18cfc5883dca07e9f9/graphlearn_torch/python/utils/common.py#L31)?

**Review by mkolodner-sc — APPROVED** [review]
Thanks!

File: gigl/common/utils/compute/random.py (hunk 2 above)
Comment (svij-sc) [reply gigl/common/utils/compute/random.py:57]: not needed, see: https://docs.pytorch.org/docs/2.12/notes/randomness.html#pytorch-random-number-generator

> You can use [torch.manual_seed()](https://docs.pytorch.org/docs/2.12/generated/torch.manual_seed.html#torch.manual_seed) to seed the RNG for all devices (both CPU and CUDA):

**Review by yliu2-sc — APPROVED** [review]
